stable/appmesh-gateway/templates/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "appmesh-gateway.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "appmesh-gateway.labels" . | indent 4 }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "appmesh-gateway.fullname" . }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "appmesh-gateway.fullname" . }}
+        app.kubernetes.io/component: "appmesh-gateway"
+      annotations:
+        {{- if .Values.appmesh.preview }}
+        appmesh.k8s.aws/preview: "enabled"
+        {{- end }}
+        {{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "appmesh-gateway.serviceAccountName" . }}
+      terminationGracePeriodSeconds: 30
+      {{- if .Values.podAntiAffinity }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app.kubernetes.io/name: {{ include "appmesh-gateway.name" . }}
+                topologyKey: kubernetes.io/hostname
+              weight: 100
+      {{- end }}
+      containers:
+      - name: envoy
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```

Comment:
Default `image` should be `injector-envoy-image` to match the [associated mutator changes](https://github.com/aws/aws-app-mesh-controller-for-k8s/commit/715b59a98280858313e5707f601b5326ca3b58cc), so the Envoy from the appmesh-manager is automatically injected.

Reply:
Correction to what Brian said: it is injected by default but you can skip it (to use user provided envoy image) using the annotation `appmesh.k8s.aws/virtualGatewaySkipImageOverride`. Here are the details [1]

[1] https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/inject/virtualgateway_envoy.go#L140

Reply:
I've added `skipImageOverride` to chart options.